modules/aql/arrow-aql/src/main/kotlin/arrow/aql/Min.kt

```diff
@@ -14,17 +14,17 @@ interface Min<F> {
 
   fun foldable(): Foldable<F>
 
-  fun <A, Y, Z> Query<F, A, Y>.min(ord: Order<Z>, f: A.() -> Z): Query<ForId, Option<Z>, Option<Z>> =
+  fun <A, Y, Z> Query<F, A, Y>.min(ord: Order<Z>, f: A.() -> Z): Query<ForId, Option<Y>, Option<Y>> =
     Query(
       select = ::identity,
       from = Id(foldable().run {
-        from.foldLeft(None) { acc: Option<Z>, a: A ->
-          acc.fold({ Some(f(a)) },
-            { Some(if (ord.run { it < f(a) }) it else f(a)) })
-        }
+        from.foldLeft(None) { acc: Option<A>, a: A ->
+          acc.fold({ Some(a) },
+            { Some(if (ord.run { f(it) < f(a) }) it else a) })
+        }.map(select)
```

Comment:
map identity? doesn't that return the same value?

Reply:
I don't follow you. I map it to select to return `Option<Y>` instead of `Option<X>`